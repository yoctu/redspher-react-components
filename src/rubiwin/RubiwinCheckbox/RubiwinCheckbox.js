import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from '@material-ui/core/Checkbox'
import { StylesProvider } from '@material-ui/core/styles'
import style from './RubiwinCheckbox.module.scss'

/** this is a Mui Checkbox branded for Rubiwin
 * full dod : https://material-ui.com/api/checkbox/
 */
const RubiwinCheckbox = ({
  checked = false,
  className = '',
  disabled = false,
  id,
  onChange,
  size = 'medium',
  inputRef,
  value,
  name
}) => (
  <StylesProvider injectFirst>
    <Checkbox
      checked={checked}
      className={`${style.checkbox} ${className}`}
      disabled={disabled}
      id={id}
      onChange={onChange}
      size={size}
      inputRef={inputRef}
      value={value}
      name={name}
      color='primary'
      classes={{
        root: style.root,
        checked: style.checked,
        disabled: style.disabled
      }}
    />
  </StylesProvider>
)

RubiwinCheckbox.propTypes = {
  /** component is checked or not */
  checked: PropTypes.bool,
  /** add a class to overload styles */
  className: PropTypes.string,
  /** disable checkbox */
  disabled: PropTypes.bool,
  /** set id of components */
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** on Change */
  onChange: PropTypes.func,
  /** size */
  size: PropTypes.oneOf(['small', 'medium']),
  /** pass a ref to the input element */
  inputRef: PropTypes.string,
  /** value of input */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** name attribute of input */
  name: PropTypes.string
}

export default RubiwinCheckbox
