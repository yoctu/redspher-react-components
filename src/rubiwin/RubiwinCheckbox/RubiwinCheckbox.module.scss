@import '../rubiwin-constant';

.checkbox {
  color: $dynamiteGrey;
}

.root {
  color: $primaryColor;
}

.checked {
  color: $primaryColor !important;
}

.disabled {
  color: $dynamiteGrey !important;
}
