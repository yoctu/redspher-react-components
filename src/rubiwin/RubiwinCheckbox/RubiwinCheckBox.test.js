import React from 'react'
import renderer from 'react-test-renderer'
import RubiwinCheckbox from './RubiwinCheckbox'

const testProps = {
  checked: false,
  className: 'testClass',
  id: 'rubiwin',
  inputRed: 'ref',
  value: 'testValue',
  name: 'rubiname',
  onChange: () => {}
}

/** Snapshots */
it('renders correctly props object', () => {
  const tree = renderer.create(<RubiwinCheckbox {...testProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is checked', () => {
  const tree = renderer.create(<RubiwinCheckbox checked />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is disabled', () => {
  const tree = renderer.create(<RubiwinCheckbox disabled />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is small', () => {
  const tree = renderer
    .create(<RubiwinCheckbox size='small' {...testProps} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
