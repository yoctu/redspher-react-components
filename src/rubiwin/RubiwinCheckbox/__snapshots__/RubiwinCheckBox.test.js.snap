// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is checked 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root root MuiCheckbox-colorPrimary checkbox  PrivateSwitchBase-checked-2 Mui-checked checked MuiIconButton-colorPrimary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={true}
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`is disabled 1`] = `
<span
  aria-disabled={true}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root root MuiCheckbox-colorPrimary checkbox  PrivateSwitchBase-disabled-3 Mui-disabled disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={false}
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
</span>
`;

exports[`is small 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root root MuiCheckbox-colorPrimary checkbox testClass MuiIconButton-colorPrimary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={false}
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      disabled={false}
      id="rubiwin"
      name="rubiname"
      onChange={[Function]}
      type="checkbox"
      value="testValue"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`renders correctly props object 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root root MuiCheckbox-colorPrimary checkbox testClass MuiIconButton-colorPrimary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={false}
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      disabled={false}
      id="rubiwin"
      name="rubiname"
      onChange={[Function]}
      type="checkbox"
      value="testValue"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
</span>
`;
